<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Exceptions implicitly reject a promise</title>
  </head>
  <body>
    <script>
      const promise = new Promise((resolve, reject) => {
        undeclaredVariable++;
      });

      promise
        .then(() => console.log("this will not run"))
        .catch((err) => console.log(err));
    </script>
  </body>
</html>
